pkgs/os-specific/linux/alsa-firmware/default.nix

```
@@ -8,6 +8,19 @@ stdenv.mkDerivation rec {
     sha256 = "0gfcyj5anckjn030wcxx5v2xk2s219nyf99s9m833275b5wz2piw";
   };
 
+  patches = [ ./cross.patch ];
+
+  nativeBuildInputs = [ automake autoconf buildPackages.stdenv.cc ];
```

```suggestion
  nativeBuildInputs = [ autoreconfHook buildPackages.stdenv.cc ];
```

 This hook does all the autotools steps, so you don't need the `preConfigure` either.

---

That is neat!